src/material/core/datetime/date-selection-model.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DateAdapter} from '@angular/material/core';
+import {Subject} from 'rxjs';
+
+export abstract class MatDateSelectionModel<D> {
+  private _valueChangesSubject = new Subject<void>();
+  valueChanges = this._valueChangesSubject.asObservable();
+
+  constructor(protected readonly adapter: DateAdapter<D>) {}
+
+  dispose() {
```

Comment:
nit: to fit in with Angular terminology, lets call this method `destroy` instead